tests/test_github_sdk.py

```diff
@@ -32,4 +39,59 @@ def test_job_basic_test(
         "https://api.github.com/repos/getsentry/sentry/actions/workflows/1174556",
         json=jobA_workflow,
     )
-    assert _generate_trace(jobA_job) == jobA_trace
+    client = GithubClient(dry_run=True)
+    assert client._generate_trace(jobA_job) == jobA_trace
+
+
+@freeze_time()
+@responses.activate
+@patch("src.github_sdk.get_uuid")
+def test_send_trace(
+    mock_get_uuid,
+    jobA_job,
+    jobA_runs,
+    jobA_workflow,
+    jobA_trace,
+    uuid_list,
+):
+    mock_get_uuid.side_effect = uuid_list
+    responses.get(
+        "https://api.github.com/repos/getsentry/sentry/actions/runs/2104746951",
+        json=jobA_runs,
+    )
+    responses.get(
+        "https://api.github.com/repos/getsentry/sentry/actions/workflows/1174556",
+        json=jobA_workflow,
+    )
+    dsn_url = "https://foo@random.ingest.sentry.io"
+    responses.post(
+        f"{dsn_url}/api/bar/envelope/",
+        json={"type": "post"},
+    )
+
+    client = GithubClient(dsn=f"{dsn_url}/bar")
+    resp = client.send_trace(jobA_job)
+    # This cannot happen in a fixture, otherwise, there will be a tiny bit of a clock drift
+    now = datetime.utcnow()
+
+    envelope_headers = {
+        "User-Agent": "python-requests/2.27.1",
+        "Accept-Encoding": "gzip, deflate",
+        "Accept": "*/*",
+        "Connection": "keep-alive",
+        "event_id": "a401d83c7ec0495f82a8da8d9a389f5b",
+        "sent_at": format_timestamp(now),
+        "Content-Type": "application/x-sentry-envelope",
+        "Content-Encoding": "gzip",
+        "X-Sentry-Auth": f"Sentry sentry_key=foo,sentry_client=gha-sentry/0.0.1,sentry_timestamp={now},sentry_version=7",
+        "Content-Length": "701",
+    }
+
+    for k, v in resp.request.headers.items():
```

Comment:
I would have wanted to compared json objects but it is some sort of custom data structure that does not have the ability to transform to a json object.